### PR TITLE
Provide a runnable mysql adapter using mysql text protocol

### PR DESCRIPTION
**Pull request by xq262144**
Implemented:
1. Runnable MySQL adapter using ```TEXT``` protocol.
2. Support ```mysql_real_connect```/```mysql_query``` using ```libmysql```.
3. Interactive with MySQL 5.x series client is supported.

Not Implemented:
1. MySQL binary protocol which often used in PHP/Python language drivers as ```mysql_prepare``` and bind parameter functions.
2. The new password authentication logic introduced in MySQL 8.0+.

P.S.: the adapter is tested and fully capable of query/show/desc tables using MySQL 5.7 command line client.

This pull request also unified all components to prefixed with ```cql-```, hope this could make the user more easy to use the project by using ```go get``` command directly.

### CODE REVIEW

**Comment by xq262144** [comment]
The MySQL binary protocol could not be implemented right now, because the MySQL ```libmysql``` client requires the server to send placeholder count and result column count during statement prepare stage.
The placeholder count in the prepared statement is possible to deliver with the help of sqlparser.
However, the result column count of the prepared statement is impossible to achieve right now, since there is no query analyzer and physical planned ahead of the storage engine.

**Comment by xq262144** [comment]
The MySQL 8+ compatibility issues will be fixed in the future.
The adapter right now is just a toy project to prove the feasibility of the protocol conversion.

File: cmd/cql-mysql-adapter/cursor.go [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+		return c.buildResultSet(rows)
+	}
+
+	var result sql.Result
+	if result, err = conn.Exec(query); err != nil {
+		err = my.NewError(my.ER_UNKNOWN_ERROR, err.Error())
+		return
+	}
+
+	lastInsertId, _ := result.LastInsertId()
```
Comment (covenantdb) [inline cmd/cql-mysql-adapter/cursor.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
var lastInsertId should be lastInsertID

File: cmd/cql-mysql-adapter/server.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	}
+
+	if s.listener, err = net.Listen("tcp", listenAddr); err != nil {
+		return
+	}
+
+	return
+}
+
+// Server starts the server.
```
Comment (covenantdb) [inline cmd/cql-mysql-adapter/server.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported method Server.Serve should be of the form "Serve ..."

**Review by leventeliu — APPROVED** [review]
LGTM, approve.

**Pull request closed without merging**